packages/tools/client-debugger/client-debugger-view/src/components/ContainerHistoryView.tsx

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Stack, StackItem } from "@fluentui/react";
+import React from "react";
+
+import { ConnectionStateChangeLogEntry } from "@fluid-tools/client-debugger";
+
+import { HasClientDebugger } from "../CommonProps";
+import { _ContainerHistoryView } from "./client-data-views";
+
+/**
+ * {@link ContainerHistory} input props.
+ */
+export type ContainerHistoryProps = HasClientDebugger;
+
+/**
+ * Displays information about the provided {@link IFluidClientDebugger.getContainerConnectionLog}.
+ *
+ * @param props - See {@link ContainerHistoryViewProps}.
+ */
+export function ContainerHistoryView(props: ContainerHistoryProps): React.ReactElement {
+	const { clientDebugger } = props;
+	const { container } = clientDebugger;
+
+	const [containerHistory, setContainerHistory] = React.useState<
+		readonly ConnectionStateChangeLogEntry[]
+	>(clientDebugger.getContainerConnectionLog());
+
+	React.useEffect(() => {
+		function onContainerHistoryChanged(): void {
+			setContainerHistory(clientDebugger.getContainerConnectionLog());
+		}
+
+		container.on("attached", onContainerHistoryChanged);
+		container.on("connected", onContainerHistoryChanged);
+		container.on("disconnected", onContainerHistoryChanged);
+		container.on("disposed", onContainerHistoryChanged);
+		container.on("closed", onContainerHistoryChanged);
+
+		return (): void => {
+			container.off("attached", onContainerHistoryChanged);
+			container.off("connected", onContainerHistoryChanged);
+			container.off("disconnected", onContainerHistoryChanged);
+			container.off("disposed", onContainerHistoryChanged);
+			container.off("closed", onContainerHistoryChanged);
+		};
+	}, [clientDebugger, container, containerHistory, setContainerHistory]);
```

Comment:
You'll want to remove `containerHistory` from this list. As written, each time the `containerHistory` value is updated, we will recreate all of the event listeners we have bound to the container, which isn't necessary.

Reply:
Removed from the list. Thx